Sources/DVR/Session.swift

```diff
@@ -181,7 +186,7 @@ open class Session: URLSession {
     }
 
     private func addTask(_ task: URLSessionTask) {
-        let shouldRecord = !recording
+        let shouldRecord = !recording && (recordMode == .newEpisodes || recordMode == .once)
         if shouldRecord {
             beginRecording()
         }
```

Comment:
Do we want to preserve this behavior where if you don't call "beginRecording" yourself it does it automagically only for that one request? Does Python behave that way? It seems wonky to me.